### PR TITLE
[Google Blockly] Gracefully fail if we can't generate code

### PR DESCRIPTION
**Pull request by molly-moen**
Previously, if `getWorkspaceCode` from the google blockly wrapper threw an exception, the entire page would crash. This can happen if we hit a bug in our code (for example, we often hit it while developing with functions and behaviors), or if the level configuration has invalid blocks. With this change, we still allow the page to load, but show the user an error message and log the error to CloudWatch so we can fix it.

### Screenshot
Screenshot 2024-02-06 at 10 05 05 AM

## Links
- jira ticket: [CT-318](https://codedotorg.atlassian.net/browse/CT-318)

## Testing story
I tested this locally with a previously broken level, fixed by [this PR](https://github.com/code-dot-org/code-dot-org/pull/56294). 

## Follow-up work
If we discover other ways blockly can crash the page, we will file follow up tasks and fix them. This change doesn't cover the case when CDO Blockly crashes due to some issue, I will investigate that as a follow up to see if it's worth fixing now.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/blockly/googleBlocklyWrapper.js [hunk 1]
```diff
@@ -160,13 +163,28 @@ function initializeBlocklyWrapper(blocklyInstance) {
...
+      if (this.getHiddenDefinitionWorkspace()) {
+        workspaceCode += Blockly.JavaScript.workspaceToCode(
+          this.getHiddenDefinitionWorkspace()
+        );
+      }
+      getStore().dispatch(setFailedToGenerateSources(false));
+    } catch (e) {
+      // We only want to log the error once per failure since getWorkspaceCode
+      // gets called many times and the error will be the same every time.
+      if (!getStore().getState().blockly.failedToGenerateSources) {
```
Comment (mikeharv) [inline apps/src/blockly/googleBlocklyWrapper.js]: Nit: Would `failedToGenerateCode` be more accurate?

**Review by mikeharv — APPROVED** [review]
Looks good!

File: apps/src/blockly/googleBlocklyWrapper.js (hunk 1 above)
Comment (molly-moen) [reply apps/src/blockly/googleBlocklyWrapper.js]: yep I'll rename

**Comment by ebeastlake** [comment]
Thanks for adding this, Molly! 🎉

**Pull request merged**